### PR TITLE
CI: update tests for numpy 1.20 change to floordiv

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Looks like https://github.com/numpy/numpy/pull/16161 changed numpy's floordiv behavior

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Yikes, I got this passing, but i have no earthly idea what is going on in the Sparse tests

File: pandas/tests/arrays/sparse/test_arithmetics.py [hunk 1]
```diff
@@ -44,9 +46,27 @@ def _check_numeric_ops(self, a, b, a_dense, b_dense, mix, op):
 
             if op in [operator.floordiv, ops.rfloordiv]:
                 # Series sets 1//0 to np.inf, which SparseArray does not do (yet)
-                mask = np.isinf(expected)
-                if mask.any():
-                    expected[mask] = np.nan
+                condition = _np_version_under1p20
+                if isinstance(a_dense, np.ndarray) and isinstance(b_dense, np.ndarray):
```
Comment (jreback) [inline pandas/tests/arrays/sparse/test_arithmetics.py:50]: can you add some comments here on what is changed, it is non-obvious from inspection what is changing (e.g. what is true in prior version and what is true now)

File: pandas/tests/arrays/sparse/test_arithmetics.py [hunk 2]
```diff
@@ -44,9 +46,27 @@ def _check_numeric_ops(self, a, b, a_dense, b_dense, mix, op):
...
+                    condition = True
+                    if a_dense.dtype == np.float64 and np.isnan(a.fill_value):
```
Comment (jreback) [inline pandas/tests/arrays/sparse/test_arithmetics.py:52]: this seems *really* fragile

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can we push an xfail for this case for now? (and debug / followup with this PR later).

**Comment by simonjayhawkins** [comment]
failing on 1.1.x as well, so will want to try to backport this.

File: pandas/tests/arrays/sparse/test_arithmetics.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/arrays/sparse/test_arithmetics.py:50]: yah as mentioned above, i have no frikkin idea what the underlying logic is here

**Comment by jreback** [comment]
agreed with the above
let's just xfail for now and backport

**Comment by jbrockmendel** [comment]
updated to xfail

File: pandas/tests/arrays/sparse/test_arithmetics.py [hunk 3]
```diff
@@ -29,7 +31,7 @@ class TestSparseArrayArithmetics:
     def _assert(self, a, b):
         tm.assert_numpy_array_equal(a, b)
 
-    def _check_numeric_ops(self, a, b, a_dense, b_dense, mix, op):
+    def _check_numeric_ops(self, a, b, a_dense, b_dense, mix, op, flag=False):
```
Comment (jreback) [inline pandas/tests/arrays/sparse/test_arithmetics.py:34]: what is flag for?

Comment (jbrockmendel) [reply pandas/tests/arrays/sparse/test_arithmetics.py:34]: leftover from previous attempt at a fix, will remove

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel  (feel free to open an issue to address this, no hurry now).

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**